### PR TITLE
Related Posts: add post classes to post's container class

### LINKED ISSUE #2809: Related Posts: add category slug to post's container class
Adding each post's category slugs to the container's class would allow site owners to style each Related Post differently, based on the attached category.

Suggested here:
https://wordpress.org/support/topic/related-posts-customizations-category-slug-class-thumbnail-size?replies=3&view=all#post-7498137

### PR DESCRIPTION
**Pull request by osiux**
Initially I was thinking on adding a new method to return only category classes, but I think it's better to use `get_post_class` since it also adds custom taxonomies and so on.

Also a new filter is there to allow users to customize the outputed css classes.

Fixes #2809

### CODE REVIEW

File: modules/related-posts/related-posts.js [hunk 1]
```diff
@@ -68,7 +68,7 @@
 
 			$.each( posts, function( index, post ) {
 				var anchor = self.getAnchor( post, 'jp-relatedposts-post-a' );
-				var classes = 'jp-relatedposts-post jp-relatedposts-post' + index;
+				var classes = 'jp-relatedposts-post jp-relatedposts-post' + index + ' ' + post.classes.join( ' ' );
```
Comment (lezama) [inline modules/related-posts/related-posts.js]: we should check if `post.classes` isn't empty before concatenating `' '`

**Comment by lezama** [comment]
This is looking good, thanks @osiux! 

I wouldn't add the post classes by default, as it could change the layout on users already using related posts.

**Comment by osiux** [comment]
@lezama yeah, i was thinking the same after sending the PR. Usually `.post` and others like that are used to style something in the default CSS of the theme. Maybe using the filter with a default empty array and let the user add whatever classes he wants? Since we are passing the post ID as parameter they can use `get_the_category()` or something else to add what they need.

**Comment by osiux** [comment]
Ok, I've updated the code. Now it checks for the empty `post.classes` in the JS and uses an empty array by default in the filter.

In the case of the suggestion post, something like this can be done:

    function add_css_classes_on_related_posts( $classes, $post_id ) {
        $categories = get_the_category( $post_id );

        foreach ( $categories as $category ) {
            $classes[] = 'category-' . $category->slug;
        }

        return $classes;
    }

    add_filter('jetpack_relatedposts_filter_post_css_classes', 'add_css_classes_on_related_posts');

**Comment by lezama** [comment]
>Maybe using the filter with a default empty array and let the user add whatever classes he wants? Since we are passing the post ID as parameter they can use get_the_category() or something else to add what they need.

That's what i meant to say :+1: 

LGTM

**Comment by dereksmart** [comment]
This looks good, @osiux - thank you!  

Can you just update the `@since` to 3.8?

**Comment by osiux** [comment]
@dereksmart done!

**Comment by mpmchugh** [comment]
Hello... I'd suggested this enhancement, so I've been following the progress here. I saw that it seemed to be done (?), so I decided to try the committed code elements and the suggested function on our dev installation.

Overall, it seems to work, but instead of adding the proper category slug for the individual related posts, it seems be just laying in the slug of current post's category on all of the related posts' divs.

Not sure if that's an issue with the code modifications, or just the function... or if this little test of mine was entirely pre-mature (other missing dependencies?), but I thought I'd report what I found.

**Comment by osiux** [comment]
@mpmchugh hello, thanks for the input! You were right, my previous example was wrong. I edited and fixed it. In short, you must call the `add_filter` function with the priority AND number of args set to 2. If that argument is missing, I guess it takes the current post as default ID for the `get_the_category` function and that's why it returns the current post's categories.

**Comment by mpmchugh** [comment]
Great. Thanks @osiux. The revised function is working perfectly on our dev server. Thanks!

Now I can't get Related Posts working at all on our WP Engine live site <sigh>, but that's not an issue for this thread.  :)

**Pull request merged**